app/src/main/java/com/example/androidtrainingproject/login/LoginActivity.kt

```diff
@@ -88,20 +87,26 @@ fun LoginScreen(navigateToProductDetails: () -> Unit = {}) {
             },
         )
 
-        if (showWrongPasswordMessage.value) {
-            Text("Invalid login credentials", color = Color.Red)
+        loginViewModel.isLoginSuccessful?.let { isLoggedIn ->
```

Comment:
Monitoring for state in a `View` this way is not a good idea. We'll talk more about more appropriate ways of doing that in the next lecture, but for now you should be able to use a callback in the login method. Hint -- you can pass the navigation method reference directly to the `login` method, we discussed this in the last lecture, or you can pass a separate callback method, whichever way it sounds better to you right now.

Reply:
This is the one part I gave up on even though I know this is bad code, I just don't know how to do this properly. I passed the navigation method to the loginViewModel